src/api_server/Cargo.toml

```diff
@@ -21,7 +21,7 @@ mmds = { path = "../mmds" }
 seccompiler = { path = "../seccompiler" }
 utils = { path = "../utils" }
 vmm = { path = "../vmm" }
-tracing = { version = "0.1.37", features = ["attributes"] }
+tracing = { git = "https://github.com/tokio-rs/tracing", rev = "27f688efb72316a26f3ec1f952c82626692c08ff", features = ["attributes"] }
```

Comment:
Why?!? I really think we should not be doing this.

Reply:
The latest version of tracing published on crates.io is 10 months ago. The main branch is up to date and includes signficant changes, I beleive this is more secure and future proof.

Reply:
> Why more secure? I would actually argue that this is less secure. Less projects are using this commit, comparing to the published version, downstream so it has gone through less testing in the field.

It is using more up to date dependencies with more up to date fixes, the majority of crates do not report when they have errors, they only fix them in version updates. In this case a more up to date version may be more secure.

In the case a security issue is found in any dependencies in the tree, we would need to do the large update from the last version onto main to resolve the issue.

> The main branch is a branch which means it can be re-written, which means this commit can disappear. I know it's very unlikely but it can happen.

If we consider this a significant risk we can fork `tracing` and depend on this fork.

> Is there a change included there that we really need and cannot implement our functionality otherwise?

I did not see the ability to re-configure the logger which underlies much of the simplification of our implementation in the older version publish to crates.io

Reply:
> In the case a security issue is found in any dependencies in the tree, we would need to do the large update from the last version onto main to resolve the issue.

Do you mean that if they find a security issue they will not cut a new release? If that's the case we should not be using this crate.

Reply:
> I was not able see the ability to re-configure the logger which underlies much of the simplification of our implementation in the older version published to crates.io.

That is problematic. Again, I think we need input from more people.

Reply:
> Do you mean that if they find a security issue they will not cut a new release? If that's the case we should not be using this crate.

No they may cut a release, but they may not cut a SVE, in this case `cargo audit` wont catch it and notify us and tracing won't publish a new version. But the latest version of tracing will always update to the latest dependencies. Thus the version on main will be safe will the older version will not be.

Reply:
That is the case with every external crate we consume. Should we point to their git history and update the hash every time the push something new?

My point is that pointing to a git has does not solve any security problem.

Reply:
Also, they may not cut as an SVE, but if they publish a new version we will get a dependabot PR for it.